src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -380,18 +343,16 @@ public IActionResult Deny()
                 });
             }
 
-            // Reject the token request if two-factor authentication has been enabled by the user.
-            if (_userManager.SupportsUserTwoFactor && await _userManager.GetTwoFactorEnabledAsync(user))
+            else if (result.RequiresTwoFactor)
             {
                 return BadRequest(new OpenIdConnectResponse
                 {
                     Error = OpenIdConnectConstants.Errors.InvalidGrant,
-                    ErrorDescription = T["The specified user is not allowed to sign in."]
+                    ErrorDescription = T["The specified user is not allowed to sign in using the password method."]
                 });
             }
 
```

Comment:
Fixed (I tend to always add blank lines between if/else if/else clauses to lighten my code :sweat_smile: )